### PR TITLE
docs - add content for new pgvector module

### PR DESCRIPTION
**Pull request by lisakowen**
docs for new pgvector module distributed with greenplum.  based on the open source module.

in this PR:
- add license file from oss repo
- create doc markdown file based on the oss README content.  to start, included most of the content from the README, reorg'd and formatted a bit differently.  would like to revise with a real use case example in the future when have more time.
  
doc review site link (behind vpn):  https://docs-staging.vmware.com/en/draft/VMware-Greenplum/pgvector/greenplum-database/GUID-ref_guide-modules-pgvector-pgvector.html

### CODE REVIEW

File: gpdb-doc/markdown/ref_guide/modules/pgvector/pgvector.html.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+The Greenplum Database `pgvector` module is equivalent to version 0.4.4 of the `pgvector` module used with PostgreSQL. The limitations of the Greenplum version of the module are described in the [Greenplum Database Limitations](#limits) topic.
+
+
+## <a id="topic_reg"></a>Installing and Registering the Module
+
+The `pgvector` module is installed when you install Greenplum Database. Before you can use the data type and index access method defined in the module, you must register the `pgvector` extension in each database in which you want to use these:
+
+```
+CREATE EXTENSION pgvector;
```
Comment (radarwave) [inline gpdb-doc/markdown/ref_guide/modules/pgvector/pgvector.html.md:13]: Should be:
CREATE EXTENSION vector;

File: gpdb-doc/markdown/ref_guide/modules/pgvector/pgvector.html.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
+# pgvector
+
+A machine language-generated embedding is a complex object transformed into a list of numbers (vector) that reflects both the semantic and syntactic relationships of the data. The `pgvector` module provides vector similarity search capabilities for Greenplum Database that enable you to search, store, and query embeddings at large scale.
+
+The Greenplum Database `pgvector` module is equivalent to version 0.4.4 of the `pgvector` module used with PostgreSQL. The limitations of the Greenplum version of the module are described in the [Greenplum Database Limitations](#limits) topic.
+
+
+## <a id="topic_reg"></a>Installing and Registering the Module
+
+The `pgvector` module is installed when you install Greenplum Database. Before you can use the data type and index access method defined in the module, you must register the `pgvector` extension in each database in which you want to use these:
```
Comment (yihong0618) [inline gpdb-doc/markdown/ref_guide/modules/pgvector/pgvector.html.md:10]: vector here maybe better?

File: gpdb-doc/markdown/ref_guide/modules/pgvector/pgvector.html.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+
+```
+CREATE EXTENSION pgvector;
+```
+
+Refer to [Installing Additional Supplied Modules](../../../install_guide/install_modules.html) for more information.
+
+
+## <a id="using"></a>About the pgvector Types, Operators, and Functions
```
Comment (yihong0618) [inline gpdb-doc/markdown/ref_guide/modules/pgvector/pgvector.html.md:19]: ditto About the vector types

File: gpdb-doc/markdown/ref_guide/modules/pgvector/pgvector.html.md (hunk 1 above)
Comment (lisakowen) [reply gpdb-doc/markdown/ref_guide/modules/pgvector/pgvector.html.md]: thanks for catching this!

**Review by yihong0618 — APPROVED** [review]
LGTM

**Review by radarwave — APPROVED** [review]

**Pull request merged**